src/test/scala/io/qbeast/spark/utils/QbeastOptimizeIntegrationTest.scala

```diff
@@ -107,4 +111,33 @@ class QbeastOptimizeIntegrationTest extends QbeastIntegrationTestSpec {
       filesAfter.map(_.blocks.size).sum should be < filesBefore.map(_.blocks.size).sum
   }
 
+  "Table optimize" should "set the dataChange flag as false" in
+    withQbeastContextSparkAndTmpDir { (spark, tmpDir) =>
+      import spark.implicits._
+
+      val df = spark.sparkContext.range(0, 10).toDF("id")
+      df.write
+        .mode("append")
+        .format("qbeast")
+        .option("columnsToIndex", "id")
+        .save(tmpDir)
+
+      QbeastTable.forPath(spark, tmpDir).optimize(1L, Map.empty[String, String])
+
+      val deltaLog = DeltaLog.forTable(spark, tmpDir)
+      val snapshot = deltaLog.update()
+      val conf = deltaLog.newDeltaHadoopConf()
+
+      val fileActions = deltaLog.store
+        .read(FileNames.deltaFile(deltaLog.logPath, snapshot.version), conf)
+        .map(Action.fromJson)
+
+      fileActions.foreach {
```

Comment:
You can also check that the `CommitInfo` has `SnapshotIsolation` as `isolationLevel`?